tests/profiling/collector/test_threading.py

```diff
@@ -82,20 +82,16 @@ def test_lock_events_tracer(tracer):
             trace_id = t.trace_id
             span_id = t.span_id
         lock2.release()
-    assert len(r.events[collector_threading.LockAcquireEvent]) == 2
-    assert len(r.events[collector_threading.LockReleaseEvent]) == 2
-    lock_event_1 = r.events[collector_threading.LockAcquireEvent][0]
-    assert lock_event_1.trace_ids is None
-    assert lock_event_1.span_ids is None
-    lock_event_2 = r.events[collector_threading.LockAcquireEvent][1]
-    assert lock_event_2.trace_ids == {trace_id}
-    assert lock_event_2.span_ids == {span_id}
-    lock_release_1 = r.events[collector_threading.LockReleaseEvent][0]
-    assert lock_release_1.trace_ids == {trace_id}
-    assert lock_release_1.span_ids == {span_id}
-    lock_release_2 = r.events[collector_threading.LockReleaseEvent][1]
-    assert lock_release_2.trace_ids is None
-    assert lock_release_2.span_ids is None
+    # The tracer might use locks, so we need to look into every event to assert we got ours
+    for event_type in (collector_threading.LockAcquireEvent, collector_threading.LockReleaseEvent):
+        assert {"test_threading.py:76", "test_threading.py:79"}.issubset({e.lock_name for e in r.events[event_type]})
```

Comment:
These hard-coded values are always a bit of a pain. Are the line numbers essential for the test to pass? If so, would it be ok to check that they are "close" enough, or at a relative distance to the first line of the test function (`test_threading.__code__.co_firstlineno`)?

Reply:
Yeah, better be safe than sorry. We don't change the test that often so it's not a big deal to maintain it. Worst case if it becomes a problem we can split those tests in their own file.